src/appmixer/ai/openai/AIAgent/AIAgent.js

```
@@ -48,20 +51,99 @@
+
+        const flowDescriptor = context.flowDescriptor;
+        const agentComponentId = context.componentId;
+        const mcpPort = 'mcp';
+        const components = {};
+        // Find all components connected to my 'mcp' output port.
+        Object.keys(flowDescriptor).forEach((componentId) => {
+            const component = flowDescriptor[componentId];
+            const sources = component.source;
+            Object.keys(sources || {}).forEach((inPort) => {
+                const source = sources[inPort];
+                if (source[agentComponentId] && source[agentComponentId].includes(mcpPort)) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)